### ISSUE TITLE
Modal unmount error on iOS Safari

### ISSUE THREAD

**Issue by samroberton** [react-materialize/react-materialize#1099]
### Description

Upgrading from react-materialize 3.4.1 to 3.8.0 caused an error in our app on iOS Safari when the Modal component unmounts.  Reverting to 3.4.1 (with no other change, and not reverting any other libraries) fixed the error.

JS console error message (sorry, we don't have a full trace):

    [Error] NotFoundError: The object can not be found here. — Modal.js:58

### Steps to Reproduce

1. Within the SPA, navigate to a page which renders a `<Modal>`.
2. Navigate to a subsequent page (which doesn't).

**Expected behavior:** navigation succeeds

**Actual behavior:** blank page, and error in the JS console as above.

### Versions

**react-materialize:** 3.8.0
**materialize-css:** customized local build
**react:** 16.13.1

**Comment by alextrastero** [comment]
Are you using SSR?

**Comment by samroberton** [comment]
> Are you using SSR?

No.  This is fully in-browser.

**Comment by alextrastero** [comment]
Can you try passing a `root` prop?

**Comment by alextrastero** [comment]
Yeah, the way storybook renders that prop is not helpful, `root` is the same proptype as `ReactDOM.render`

`ReactDOM.render(<App />, document.getElementById('root'))`

So for the Modal:

```jsx
<Modal
  root={document.getElementById('root')}
  ...
```

I hope this is helpful
